products/cms-integration/transformation-process.md

```diff
@@ -0,0 +1,254 @@
+
+# Table of Contents
+
+1.  [Transformation Process](#org6b499b3)
+    1.  [The actors](#org70bbb4e)
+        1.  [Content models](#orgd9cac48)
+        2.  [Raw content](#orgb12cb7b)
+        3.  [Transformed content](#org07d25fc)
+        4.  [Transformer](#org3a44eb7)
+        5.  [Liquid templates](#org187853e)
+    2.  [The process](#org48df923)
+        1.  [Metalsmith pipeline](#org0b02218)
+        2.  [CMS export tarball](#org458259d)
+        3.  [Pre-transformation JSON schema](#org187b1ae)
+        4.  [Content model transformers](#orgf314d86)
+        5.  [Post-transformation JSON](#org0fada70)
+        6.  [Liquid templates](#orgf5df85e)
+    3.  [Testing](#org3becc44)
+    4.  [Digging deep](#orgc84a542)
+        1.  [Automatic schema imports](#orgd8989fc)
+        2.  [Automatic `$id`](#orga2b24c5)
+    5.  [Debugging](#orgc67539d)
+
+
+
+<a id="org6b499b3"></a>
+
+# Transformation Process
+
+So you want to learn some black magic, eh? Please, step inside. I&rsquo;ll show you
+how the sausage is made.
+
+
+<a id="org70bbb4e"></a>
+
+## The actors
+
+The goal of the whole transformer system is to change all the content models
+from one shape to another. In the end, it&rsquo;s just inputs (raw content), process
+(the transformers), and outputs (transformed content).
+
+Before we talk about how it all works, let&rsquo;s get some shared terminology first.
+
+
+<a id="orgd9cac48"></a>
+
+### Content models
+
+A content model has a two-part name, for the purposes of this integration:
+
+1.  A base type
+2.  A sub type
+
+The **base type** is found in the first part of the name of the file in the CMS
+export content directory. For a `node.<UUID>.json` file, its base type is
+`node`. See the section [Raw content](#orgb12cb7b) below for context.
+
+The **sub type** is found *inside* the entity, usually in its `type[0].target_id`
+property.
+
+To get the full name, simply separate them with a hyphen. Examples include
+`node-page` and `paragraph-q_a_section`.
+
+
+<a id="orgb12cb7b"></a>
+
+### Raw content
+
+We start with the raw content as found in the CMS export tarball. When
+extracted, this tarball creates a directory with a child `content/`. In *there*
+are thousands of flat JSON files and a `meta/index.json`, but we can ignore that
+one for now.
+
+Each file is named like `node.<UUID>.json` where `<UUID>` is the UUID of the
+entity.
+
+
+<a id="org07d25fc"></a>
+
+### Transformed content
+
+After content has run through a [transformer](#org3a44eb7), we call it `transformed content`.
```

Comment:
Suggest using simpler sentence structures. 

```suggestion
The [transformer](#org3a44eb7) converts content into `transformed content` 
```

Reply:
I guess the above could describe the transformer and you could describe this as "The output of a transformer."